components/eamxx/src/share/util/scream_universal_constants.hpp

```diff
@@ -1,13 +1,22 @@
 #ifndef SCREAM_UNIVERSAL_CONSTANTS_HPP
 #define SCREAM_UNIVERSAL_CONSTANTS_HPP
 
+#include <limits>
+
 namespace scream {
 
 namespace constants {
 
 constexpr int seconds_per_day       = 86400;
 constexpr int days_per_nonleap_year = 365;
 
+// Universal fill value for variables
+// TODO: When we switch to supporting C++17 we can use a simple `inline constexpr` rather than a struct
+template<typename T>
+struct DefaultFillValue {
```

Comment:
@bartgol , the last fail for `debug_nopack_fpe` is w.r.t. this line when constructing DefaultFillValue with `T` of `integer` type.  The numeric limits for a `float` are not representable as an `int`.  

We don't want to use `std::numeric_limits<T>::max()` because of the issue with `double` precision and our single precision output.  But the std::max for `int` is pretty small compared with float and double, so maybe we don't want to use that as the default for all either.  Any thoughts on what to do?

Reply:
Perhaps you can use `std::numeric_limits<T>::max()` with T=float for float/double and T=int for all other types. You could do this

```
template<typename T>
struct DefaultFillValue {
   using fill_type = std::conditional<std::is_integral<T>::value,
                                                        int,
                                                        std::conditional<std::is_floating_point<T>::value,float, char>::type
                                                       >::type;
   static constexpr fill_type value =  std::numeric_limits<type>::max();
};
```
This will use `int` for any integral type (will work without overflow also for `int64_t`), `float` for any floating point type (no overflow for `double`), and will use `char` otherwise.

Reply:
okay great! Thanks for the suggestion.  I'll implement that now

Reply:
I had to adjust what you pasted a little bit because I was hitting compiler errors, but what I have now should work.